src/main/java/bio/terra/flight/snapshot/create/CreateSnapshotFireStoreDataStep.java

```diff
@@ -0,0 +1,100 @@
+package bio.terra.flight.snapshot.create;
+
+import bio.terra.filesystem.FireStoreDao;
+import bio.terra.filesystem.FireStoreDependencyDao;
+import bio.terra.metadata.Dataset;
+import bio.terra.metadata.Snapshot;
+import bio.terra.metadata.SnapshotMapColumn;
+import bio.terra.metadata.SnapshotMapTable;
+import bio.terra.metadata.SnapshotSource;
+import bio.terra.model.SnapshotRequestModel;
+import bio.terra.pdao.bigquery.BigQueryPdao;
+import bio.terra.service.DatasetService;
+import bio.terra.service.SnapshotService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+
+public class CreateSnapshotFireStoreDataStep implements Step {
+
+    private BigQueryPdao bigQueryPdao;
+    private SnapshotService snapshotService;
+    private FireStoreDependencyDao dependencyDao;
+    private DatasetService datasetService;
+    private SnapshotRequestModel snapshotReq;
+    private FireStoreDao fileDao;
+
+    public CreateSnapshotFireStoreDataStep(BigQueryPdao bigQueryPdao,
+                                           SnapshotService snapshotService,
+                                           FireStoreDependencyDao dependencyDao,
+                                           DatasetService datasetService,
+                                           SnapshotRequestModel snapshotReq,
+                                           FireStoreDao fileDao) {
+        this.bigQueryPdao = bigQueryPdao;
+        this.snapshotService = snapshotService;
+        this.dependencyDao = dependencyDao;
+        this.datasetService = datasetService;
+        this.snapshotReq = snapshotReq;
+        this.fileDao = fileDao;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) {
+        // We need a complete snapshot; use the snapshotService to get one.
+        Snapshot snapshot = snapshotService.retrieveSnapshotByName(snapshotReq.getName());
+
+        // Build the snapshot file system and record the file dependencies
+        // The algorithm is:
+        // Loop through sources, loop through map tables, loop through map columns
+        // if from column is FILEREF or DIRREF, ask pdao to get the ids from that
+        // column that are in the snapshot; do the filestore processing
+        //
+        // NOTE: This is brute force doing a column at a time. Depending on how much memory and
+        // swap we want to use, we could extract all row ids in one go. Doing it column-by-column
+        // bounds the intermediate size in a way. I think this all becomes easier if we move
```

Comment:
The end of this comment is a little out of date :)

Reply:
Indeed. Removed in DR-555